aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,296 @@
+:version: 1.0.0
+
+Hosting
+=======
+
+By `Steve Smith`_
+
+To run an ASP.NET Core app, you need to configure and launch a host using `WebHostBuilder <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Hosting/WebHostBuilder/index.html?highlight=webhostbuilder>`_.
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+What is a Host?
+---------------
+
+ASP.NET Core apps require a *host* in which to execute. A host must implement the :dn:iface:`~Microsoft.AspNetCore.Hosting.IWebHost` interface, which exposes collections of features and services, and a ``Start`` method. The host is typically created using an instance of a :dn:class:`~Microsoft.AspNetCore.Hosting.WebHostBuilder`, which builds and returns a  :dn:class:`~Microsoft.AspNetCore.Hosting.Internal.WebHost` instance. The ``WebHost`` has a `private ``Server`` property <https://github.com/aspnet/Hosting/blob/1.0.0/src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs#L40>`_ that specifies the server that will handle requests. Learn more about :doc:`servers <servers>`.
+
+What is the difference between a host and a server?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The host is responsible for ensuring the application's services and the server are available and properly configured. You can think of the host as being a wrapper around the server. The host is configured to use a particular server; the server is unaware of its host.
+
+Setting up a Host
+-----------------
+
+You create a host using an instance of ``WebHostBuilder``. This is typically done in your app's entry point, which by default will be located in a *Program.cs* file. A typical *Program.cs*, shown below, demonstrates how to use a ``WebHostBuilder`` to build a host. 
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Program.cs
+  :emphasize-lines: 14-21
+  :language: c#
+  :caption: Program.cs
+
+The ``WebHostBuilder`` is responsible for creating the host that will bootstrap the server for the app. ``WebHostBuilder`` requires you provide a server that implements :dn:iface:`~Microsoft.AspNetCore.Hosting.Server.IServer` (``UseKestrel`` in the code above). ``UseKestrel`` specifies the built-in Kestrel server will be used by the app.
+
+The server's *content root* determines where it searches for content files, like MVC View files. The default content root is the folder from which the application is run. 
+
+.. note:: Specifying ``Directory.GetCurrentDirectory`` as the content root will use the web project's root folder as the app's content root when the app is started from this folder (for example, calling ``dotnet run`` from the web project folder). This is the default used in Visual Studio and ``dotnet new`` templates.
```

Comment:
@Tratcher  can we say something like 
`.UseContentRoot(Directory.GetCurrentDirectory())`
will work in most environments (Azure, VS, dotnet net). 
Do we want to say why/when you would change this?